src/metabase/lib/dev.cljc

```diff
@@ -0,0 +1,49 @@
+(ns metabase.lib.dev
+  "Conveniences for usage in REPL and tests. Things in this namespace are not meant for normal usage in the FE client or
+  in QB code."
+  (:require
+   [metabase.lib.field :as lib.field]
+   [metabase.lib.metadata :as lib.metadata]
+   [metabase.lib.query :as lib.query]
+   [metabase.lib.schema :as lib.schema]
+   [metabase.lib.schema.common :as lib.schema.common]
+   [metabase.lib.schema.id :as lib.schema.id]
+   [metabase.util.malli :as mu]))
+
+(mu/defn field :- fn?
+  "Returns a function that can get resolved to an actual `:field` clause later."
+  ([id-or-name :- [:or
+                   ::lib.schema.id/field
+                   ::lib.schema.common/non-blank-string]]
+   (if (integer? id-or-name)
+     (let [id id-or-name]
+       (fn [query stage-number]
+         (->> (lib.metadata/field query id)
```

Comment:
I think all of these rules are more of guidelines and not strict rules that can never be broken.

I can't remember which linter it was, but I've definitely seen one that could be configured to complain about `->` or `->>` with only two args, because you could just pass the first form directly to the second and avoid the threading altogether.

But I think in some cases the tasteful usage of threading macros make things more readable. I think this is one of those cases. The code is clearer with the threading macro. 

I think the most important rule is do whatever makes the code easiest to read